awscli/clidriver.py

```diff
@@ -220,7 +221,13 @@ def main(self, args=None):
             LOG.debug("Exception caught in main()", exc_info=True)
             LOG.debug("Exiting with rc 255")
             sys.stderr.write("\n")
-            sys.stderr.write("%s\n" % e)
+            if six.PY3:
```

Comment:
So usually if there is a python 2/3 compatibility logic, the typical pattern in the CLI codebase is to add it to the ``compat`` module. Digging through the ``compat`` module, I found this commit that James made: https://github.com/aws/aws-cli/commit/8bb3f1aa9ab52f5dcb57e59cb45ff9b83c543457. So it looks like we have ran into this issue before, but with standard out. So it may be worth just copying that pattern or expanding the compat method to a general ``get_text_wrapper()`` function that you can provide stdin or stdout to.